### ISSUE TITLE
How do we plan to evaluate new hires?

### ISSUE THREAD

**Issue by greghaskins** [PillarTechnology/guild-docs#16]
Does everybody start at Apprentice? Or do past employers/clients/experience pre-Pillar count for something?

**Comment by jameskbride** [comment]
Similar question, but related to the number of years of experience, and the number of clients an individual should have been at.  Are we counting experience prior to joining Pillar?

**Comment by oshbec** [comment]
We should be able to use this to evaluate new hires in the same way we evaluate current team members.  Most bullet points are applicable to both, save for the piece specifying number of *Pillar* projects.

To be a Craftsman as a current employee, you need to have worked on 3 Pillar projects.  But as a prospective Craftsman hire, you haven't worked on any *Pillar* projects before.  Still, we would probably expect you to have worked on a few (3?) projects of some significance and to be able to articulate your contributions.

You might also join Pillar as a mid-level Journeyman.  Your past experience and skill level may mean you only need to be on 1-2 more projects before you're considered to be Craftsman-level.

In short, outside projects should count.  @jeffhoover points out that if we are counting outside / previous projects, they should be sufficiently Pillar-like (TDD, CI, collective code ownership, team room, etc) in order to be considered.

**Comment by JeffHoover** [comment]
I don't think starting new experienced hires at Apprentice is workable. Here's why:

If every new hire were to join Pillar with the Apprentice label, would everyone also join at Apprentice salary? (acknowledging that there may not be a 1:1 relationship between "role" and pay) If everyone joined at Apprentice salary, then we would need to have an extremely effective and efficient path to pay raises, so that new hires could trust that their "below market" salary would be *very* temporary. 

For new experienced hires starting at Apprentice, would Pillar would bill the client at Apprentice rates ? I'd have to guess no, as it is difficult to raise rates on a client once a person is on the project. A new hire initially being paid Apprentice salary may find it unfair that, until they are brought up to market salary rates, Pillar would be making a much-larger-than-usual margin on them by billing the client at rates for an experienced person.

While there are employers that people are willing to join at "below market" salaries, I don't know if Pillar is one yet. So we could miss out on some good hires by starting experienced people at Apprentice.

**Comment by jfolkerth** [comment]
I would recommend an honestly communicated probationary period for new hires at the non-Apprentice level. This lets people be hired in at what their actual title/pay would be, prevents weirdness in client billing, and allows time for those unused to our practices to learn and start using them.

Alternatively, we could only hire people who demonstrate capability with our coding practices (or at least the ability to adapt to them quickly) as appropriate for the desired persona. I doubt the market would support this given our desired growth rate, so this would necessitate more Apprentice and Journeyman level hiring and dedicated effort to train such individuals in our ways.

I would be wary of being too strict about the "Pillar-like" nature of previous projects as that is usually out of a person's control, much like how many clients/projects they have served on. It would probably be more effective to evaluate what their capabilities are throughout the hiring and probationary period to get a sense for what their skills are right now, rather than what the technology practices of their previous/current employers were like.

In general, I like the approach of articulating what behaviors and skills are being sought for, rather than using hard coded numbers of years and clients. Certain skills are going to be harder to acquire without having a diversity of experiences, but those experiences can come from things besides professional work. Looking at the person rather than the source of their skills seems more likely to yield positive results.

**Comment by JeffHoover** [comment]
I agree with @jfolkerth that I prefer "articulating what behaviors and skills are being sought for, rather than ... hard coded numbers of years and clients." and I think some measurable metrics are helpful, to avoid entirely subjective assessments of a person's progress.
